flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c

```diff
@@ -333,42 +333,50 @@ static void _msp_send_altitude(struct msp_bridge *m)
 	msp_send(m, MSP_ALTITUDE, data.buf, sizeof(data));
 }
 
+static uint16_t _msp_scale_rc(ActuatorSettingsData actuatorSettings,
+			      ManualControlCommandData manualState,
+			      int channel_num)
+{
+	uint16_t val = manualState.Channel[channel_num];
+	uint16_t min = actuatorSettings.ChannelMin[channel_num];
+	uint16_t max = actuatorSettings.ChannelMax[channel_num];
+	uint16_t idle = actuatorSettings.ChannelNeutral[channel_num];
+
+	// Min and max may be reversed.  If the value is outside of
+	// the given range, return the neutral value.
+	uint16_t lower = min < max ? min : max;
+	uint16_t higher = max > min ? max : min;
+	if (val > lower || val < higher) {
+		return idle;
+	}
+
+	double d = 1000.0 / ((double)max - (double)min);
+	double off = val - min;
+	return 1000 + (d * off);
+}
+
+// MSP RC order is Roll/Pitch/Yaw/Throttle/AUX1/AUX2/AUX3/AUX4
 static void _msp_send_channels(struct msp_bridge *m)
 {
 	ManualControlCommandData manualState;
 	ManualControlCommandGet(&manualState);
-
-	// MSP RC order is Roll/Pitch/Yaw/Throttle/AUX1/AUX2/AUX3/AUX4
-	static const uint8_t channel_map[] = {
-		MANUALCONTROLCOMMAND_CHANNEL_ROLL,
-		MANUALCONTROLCOMMAND_CHANNEL_PITCH,
-		MANUALCONTROLCOMMAND_CHANNEL_YAW,
-		MANUALCONTROLCOMMAND_CHANNEL_THROTTLE,
-		MANUALCONTROLCOMMAND_CHANNEL_FLIGHTMODE,
-		MANUALCONTROLCOMMAND_CHANNEL_ACCESSORY0,
-		MANUALCONTROLCOMMAND_CHANNEL_ACCESSORY1,
-		MANUALCONTROLCOMMAND_CHANNEL_ACCESSORY2,
-	};
+	ActuatorSettingsData actuatorSettings;
+	ActuatorSettingsGet(&actuatorSettings);
```

Comment:
As the settings don't change frequently, you could use a global variable for this and use a callback to update it. This reduces CPU usage but uses more memory (instead using stack, the variable is allocated globally and some memory is used for the callback).